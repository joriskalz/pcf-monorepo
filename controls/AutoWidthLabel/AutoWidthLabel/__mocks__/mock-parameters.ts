/* istanbul ignore file */

import { IInputs } from "../generated/ManifestTypes";
import {
  MockDecimalNumberProperty,
  MockEnumProperty,
  MockStringProperty,
  MockWholeNumberProperty,
} from "./mock-context";

export function getMockParameters(): IInputs {
  return {
    Text: new MockStringProperty(),
    PaddingLeft: new MockWholeNumberProperty(),
    PaddingRight: new MockWholeNumberProperty(),
    PaddingTop: new MockWholeNumberProperty(),
    PaddingBottom: new MockWholeNumberProperty(),
    FontName: new MockStringProperty(),
    FontSize: new MockDecimalNumberProperty(),
    FontSizeUnits: new MockEnumProperty("1"),
    FontColor: new MockStringProperty(),
    FontWeight: new MockStringProperty(),
    DisabledFontColor: new MockStringProperty(),
    DisabledFontWeight: new MockStringProperty(),
    FocusFontColor: new MockStringProperty(),
    FocusFontWeight: new MockStringProperty(),
    HoverFontColor: new MockStringProperty(),
    HoverFontWeight: new MockStringProperty(),
    FillColor: new MockStringProperty(),
    DisabledFillColor: new MockStringProperty(),
    FocusFillColor: new MockStringProperty(),
    HoverFillColor: new MockStringProperty(),
    //DisabledHoverFillColor: new MockStringProperty(),
    BorderColor: new MockStringProperty(),
    BorderThickness: new MockWholeNumberProperty(),
    BorderRadius: new MockWholeNumberProperty(),
    DisabledBorderColor: new MockStringProperty(),
    FocusBorderColor: new MockStringProperty(),
    FocusBorderThickness: new MockWholeNumberProperty(),
    HoverBorderColor: new MockStringProperty(),
    HoverBorderThickness: new MockWholeNumberProperty(),
    //RenderBorderStyle: new MockEnumProperty<"0" | "1">(),
  };
}
