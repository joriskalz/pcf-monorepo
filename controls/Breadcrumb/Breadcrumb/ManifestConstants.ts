export const enum ManifestPropertyNames {
    dataset = 'dataset',
}

export const enum ItemColumns {
    DisplayName = 'ItemDisplayName',
    Key = 'ItemKey',
    Clickable = 'ItemClickable',
}

export const enum InputEvents {
    SetFocus = 'SetFocus',
}

export const enum InputProperties {
    InputEvent = 'InputEvent',
    SelectedKey = 'SelectedKey',
}
