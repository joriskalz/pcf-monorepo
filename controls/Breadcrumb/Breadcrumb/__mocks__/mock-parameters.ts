/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockStringProperty, MockWholeNumberProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        AccessibilityLabel: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        Theme: new MockStringProperty(),
        items: new MockDataSet([]),
        MaxDisplayedItems: new MockWholeNumberProperty(),
        OverflowIndex: new MockWholeNumberProperty(),
    };
}
