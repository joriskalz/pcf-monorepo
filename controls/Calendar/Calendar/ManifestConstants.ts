export const enum InputEvents {
    SetFocus = 'SetFocus',
}
