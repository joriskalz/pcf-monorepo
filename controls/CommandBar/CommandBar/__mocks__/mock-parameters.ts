/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockStringProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        Theme: new MockStringProperty(),
        AccessibilityLabel: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        items: new MockDataSet([]),
    };
}
