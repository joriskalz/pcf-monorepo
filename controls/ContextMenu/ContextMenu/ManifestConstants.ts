export const enum ManifestPropertyNames {
    dataset = 'dataset',
}

export const enum ItemColumns {
    DisplayName = 'ItemDisplayName',
    Key = 'ItemKey',
    Enabled = 'ItemEnabled',
    IconName = 'ItemIconName',
    IconColor = 'ItemIconColor',
    ParentKey = 'ItemParentKey',
    IconOnly = 'ItemIconOnly',
    Visible = 'ItemVisible',
    Checked = 'ItemChecked',
    ItemHeader = 'ItemHeader',
    ItemTopDivider = 'ItemTopDivider',
    ItemDivider = 'ItemDivider',
}

export const enum InputEvents {
    SetFocus = 'SetFocus',
}

export const enum InputProperties {
    InputEvent = 'InputEvent',
}

export const TextAlignmentTypes = {
    '0': 'center',
    '1': 'left',
    '2': 'right',
};
