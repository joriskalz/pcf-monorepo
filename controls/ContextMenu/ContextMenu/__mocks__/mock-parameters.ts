/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockEnumProperty, MockStringProperty, MockTwoOptionsProperty, MockWholeNumberProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        AccessibilityLabel: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        Theme: new MockStringProperty(),
        TextAlignment: new MockEnumProperty(),
        items: new MockDataSet([]),
        Chevron: new MockTwoOptionsProperty(),
        FillColor: new MockStringProperty(),
        BorderColor: new MockStringProperty(),
        BorderRadius: new MockWholeNumberProperty(),
        FontColor: new MockStringProperty(),
        FontSize: new MockWholeNumberProperty(),
        HoverBorderColor: new MockStringProperty(),
        HoverFillColor: new MockStringProperty(),
        HoverFontColor: new MockStringProperty(),
        HoverIconColor: new MockStringProperty(),
        IconColor: new MockStringProperty(),
        IconSize: new MockWholeNumberProperty(),
    };
}
