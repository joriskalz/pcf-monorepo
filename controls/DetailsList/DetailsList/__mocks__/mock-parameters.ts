/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockEnumProperty, MockStringProperty, MockTwoOptionsProperty, MockWholeNumberProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        AccessibilityLabel: new MockStringProperty(),
        InputEvent: new MockStringProperty(),
        Theme: new MockStringProperty(),
        Compact: new MockTwoOptionsProperty(),
        CurrentSortColumn: new MockStringProperty(),
        CurrentSortDirection: new MockEnumProperty('0'),
        PageSize: new MockWholeNumberProperty(),
        RaiseOnRowSelectionChangeEvent: new MockTwoOptionsProperty(),
        SelectRowsOnFocus: new MockTwoOptionsProperty(),
        SelectionType: new MockEnumProperty('0'),
        AlternateRowColor: new MockStringProperty(),
        HeaderVisible: new MockTwoOptionsProperty(false),
        SelectionAlwaysVisible: new MockTwoOptionsProperty(false),
        records: new MockDataSet([]),
        columns: new MockDataSet([]),
        LargeDatasetPaging: new MockTwoOptionsProperty(),
    };
}
