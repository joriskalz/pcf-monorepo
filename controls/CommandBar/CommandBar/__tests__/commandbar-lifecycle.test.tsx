/* eslint-disable sonarjs/no-identical-functions */
import { mount } from 'enzyme';
import ReactDOM from 'react-dom';
import * as ReactTestUtils from 'react-dom/test-utils';
import { CommandBar } from '..';
import { ContextEx } from '../ContextExtended';
import { IInputs } from '../generated/ManifestTypes';
import { ItemColumns } from '../ManifestConstants';
import { MockContext, MockState } from '../__mocks__/mock-context';
import { MockDataSet, MockEntityRecord } from '../__mocks__/mock-datasets';
import { getMockParameters } from '../__mocks__/mock-parameters';

// Since requestAnimationFrame does not exist in the test DOM, mock it
window.requestAnimationFrame = jest.fn().mockImplementation((callback) => {
    callback();
});

const runAllTimers = () =>
    ReactTestUtils.act(() => {
        jest.runAllTimers();
    });

describe('CommandBar', () => {
    beforeEach(() => {
        jest.clearAllMocks();
        jest.useFakeTimers();
    });
    afterEach(() => {
        for (let i = 0; i < document.body.children.length; i++) {
            if (document.body.children[i].tagName === 'DIV') {
                document.body.removeChild(document.body.children[i]);
                i--;
            }
        }
        // reset any jest timers
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        if ((setTimeout as any).mock) {
            jest.runOnlyPendingTimers();
            jest.useRealTimers();
        }
    });

    it('renders', () => {
        const { component, context } = createComponent();
        component.init(context);
        const element = component.updateView(context);
        expect(element).toMatchSnapshot();
    });

    it('renders dummy items when no items configured', () => {
        const { component, context } = createComponent();
        // Simulate there being no items bound - which causes an error on the parameters
        context.parameters.items.error = true;
        component.init(context);
        const element = component.updateView(context);
        expect(element).toMatchSnapshot();
    });

    it('raises the onSelect event', () => {
        const { component, context } = createComponent();
        component.init(context);
        const commandBarElement = component.updateView(context);

        const firstCommandReference = {
            id: { guid: '1' },
            name: '1',
        } as ComponentFramework.EntityReference;
        context.parameters.items.records['1'].getNamedReference = jest.fn().mockReturnValueOnce(firstCommandReference);

        const commandBar = mount(commandBarElement);
        const menuItem = commandBar.find('button.ms-Button').first();
        expect(menuItem.length).toEqual(1);
        menuItem.simulate('click');
        expect(context.parameters.items.openDatasetItem).toBeCalledTimes(1);
        expect(context.parameters.items.openDatasetItem).toBeCalledWith(firstCommandReference);
    });

    it('renders subitems and shows the flyout menu', () => {
        const { component, context } = createComponent(true);

        component.init(context);
        const commandBarElement = component.updateView(context);
        const firstCommandReference = {
            id: { guid: '1' },
            name: '1',
        } as ComponentFramework.EntityReference;
        context.parameters.items.records['1'].getNamedReference = jest.fn().mockReturnValueOnce(firstCommandReference);

        // The sub items should not be included in the root commands
        expect(commandBarElement).toMatchSnapshot();

        const commandBar = mount(commandBarElement);
        const menuItem = commandBar.find('button.ms-Button').first();
        expect(menuItem.length).toEqual(1);

        menuItem.simulate('click');

        const subItem = document.querySelector('.ms-ContextualMenu-itemText');
        expect(subItem).toBeDefined();
        expect(subItem?.childNodes[0].textContent).toBe('Sub 1');
    });

    it('sets focus on control', () => {
        expect.assertions(1);
        const { component, context, container } = createComponent();

        ReactTestUtils.act(() => {
            component.init(context);
            ReactDOM.render(component.updateView(context), container);
        });

        context.parameters.InputEvent.raw = 'SetFocus' + Math.random().toString();

        ReactTestUtils.act(() => {
            ReactDOM.render(component.updateView(context), container);
        });

        runAllTimers();

        expect(document.activeElement?.tagName).toBe('BUTTON');

        ReactDOM.unmountComponentAtNode(container);
    });

    it('does not cause duplicate key warnings if the ItemKey is duplicated', () => {
        jest.spyOn(console, 'error').mockImplementation();
        const { component, context } = createComponent();

        context.parameters.items = new MockDataSet([
            new MockEntityRecord('1', {
                [ItemColumns.Key]: 'item1',
                [ItemColumns.DisplayName]: 'Open',
            }),
            new MockEntityRecord('2', {
                [ItemColumns.Key]: 'item1',
                [ItemColumns.DisplayName]: 'New',
            }),
        ]);
        component.init(context);

        // Check there is no duplicate key message
        mount(component.updateView(context));

        // We check that there are no 'Warning: Encountered two children with the same key' messages
        expect(console.error).toHaveBeenCalledTimes(0);
    });
});

function createComponent(subCommands?: boolean) {
    const component = new CommandBar();
    const notifyOutputChanged = jest.fn();
    const context = new MockContext<IInputs>(getMockParameters());
    (context as unknown as ContextEx).accessibility.assignedTabIndex = 5;
    const items = [
        new MockEntityRecord('1', {
            [ItemColumns.Key]: 'commandOpen',
            [ItemColumns.DisplayName]: 'Open',
            [ItemColumns.IconName]: 'Open',
            [ItemColumns.IconColor]: 'blue',
            [ItemColumns.Enabled]: true,
            [ItemColumns.IconOnly]: false,
        }),
        new MockEntityRecord('2', {
            [ItemColumns.Key]: 'commandNew',
            [ItemColumns.DisplayName]: 'New',
            [ItemColumns.IconName]: 'New',
            [ItemColumns.IconColor]: 'blue',
            [ItemColumns.Enabled]: false,
            [ItemColumns.IconOnly]: true,
        }),
        new MockEntityRecord('3', {
            [ItemColumns.Key]: 'commandSave',
            [ItemColumns.DisplayName]: 'Save',
            [ItemColumns.IconName]: 'Save',
            [ItemColumns.IconColor]: 'green',
            [ItemColumns.Enabled]: false,
            [ItemColumns.IconOnly]: true,
            [ItemColumns.FarItem]: true,
        }),
    ];
    if (subCommands) {
        items.push(
            new MockEntityRecord('5', {
                [ItemColumns.Key]: 'sub1',
                [ItemColumns.DisplayName]: 'Sub 1',
                [ItemColumns.IconName]: 'New',
                [ItemColumns.ParentKey]: 'commandOpen',
            }),
        );
        items.push(
            new MockEntityRecord('6', {
                [ItemColumns.Key]: 'sub2',
                [ItemColumns.DisplayName]: 'Sub 2',
                [ItemColumns.IconName]: 'Save',
                [ItemColumns.ParentKey]: 'commandOpen',
            }),
        );
    }
    context.parameters.items = new MockDataSet(items);
    const state = new MockState();
    const container = document.createElement('div');
    document.body.appendChild(container);
    return { component, context, container, notifyOutputChanged, state };
}
