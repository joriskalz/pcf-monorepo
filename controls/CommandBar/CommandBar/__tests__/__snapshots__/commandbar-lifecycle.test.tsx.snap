// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommandBar renders 1`] = `
<Memo(CanvasCommandBar)
  disabled={false}
  height={-1}
  items={
    Array [
      Object {
        "checked": undefined,
        "data": MockEntityRecord {
          "getNamedReference": [MockFunction],
          "id": "1",
          "values": Object {
            "ItemDisplayName": "Open",
            "ItemEnabled": true,
            "ItemIconColor": "blue",
            "ItemIconName": "Open",
            "ItemIconOnly": false,
            "ItemKey": "commandOpen",
          },
        },
        "divider": undefined,
        "enabled": true,
        "farItem": undefined,
        "iconColor": "blue",
        "iconName": "Open",
        "iconOnly": false,
        "id": "1",
        "isHeader": undefined,
        "key": "commandOpen",
        "name": "Open",
        "overflow": undefined,
        "parentItemKey": undefined,
        "split": undefined,
        "topDivider": undefined,
        "visible": undefined,
      },
      Object {
        "checked": undefined,
        "data": MockEntityRecord {
          "getNamedReference": [MockFunction],
          "id": "2",
          "values": Object {
            "ItemDisplayName": "New",
            "ItemEnabled": false,
            "ItemIconColor": "blue",
            "ItemIconName": "New",
            "ItemIconOnly": true,
            "ItemKey": "commandNew",
          },
        },
        "divider": undefined,
        "enabled": false,
        "farItem": undefined,
        "iconColor": "blue",
        "iconName": "New",
        "iconOnly": true,
        "id": "2",
        "isHeader": undefined,
        "key": "commandNew",
        "name": "New",
        "overflow": undefined,
        "parentItemKey": undefined,
        "split": undefined,
        "topDivider": undefined,
        "visible": undefined,
      },
      Object {
        "checked": undefined,
        "data": MockEntityRecord {
          "getNamedReference": [MockFunction],
          "id": "3",
          "values": Object {
            "ItemDisplayName": "Save",
            "ItemEnabled": false,
            "ItemFarItem": true,
            "ItemIconColor": "green",
            "ItemIconName": "Save",
            "ItemIconOnly": true,
            "ItemKey": "commandSave",
          },
        },
        "divider": undefined,
        "enabled": false,
        "farItem": true,
        "iconColor": "green",
        "iconName": "Save",
        "iconOnly": true,
        "id": "3",
        "isHeader": undefined,
        "key": "commandSave",
        "name": "Save",
        "overflow": undefined,
        "parentItemKey": undefined,
        "split": undefined,
        "topDivider": undefined,
        "visible": undefined,
      },
    ]
  }
  onSelected={[Function]}
  setFocus=""
  tabIndex={5}
  themeJSON={null}
  visible={true}
  width={-1}
/>
`;

exports[`CommandBar renders dummy items when no items configured 1`] = `
<Memo(CanvasCommandBar)
  disabled={false}
  height={-1}
  items={
    Array [
      Object {
        "iconName": "Unknown",
        "id": "1",
        "key": "1",
        "name": "Item 1",
      },
      Object {
        "iconName": "Unknown",
        "id": "2",
        "key": "2",
        "name": "Item 2",
      },
      Object {
        "iconName": "Unknown",
        "id": "3",
        "key": "3",
        "name": "Item 3",
      },
    ]
  }
  onSelected={[Function]}
  setFocus=""
  tabIndex={5}
  themeJSON={null}
  visible={true}
  width={-1}
/>
`;

exports[`CommandBar renders subitems and shows the flyout menu 1`] = `
<Memo(CanvasCommandBar)
  disabled={false}
  height={-1}
  items={
    Array [
      Object {
        "checked": undefined,
        "data": MockEntityRecord {
          "getNamedReference": [MockFunction],
          "id": "1",
          "values": Object {
            "ItemDisplayName": "Open",
            "ItemEnabled": true,
            "ItemIconColor": "blue",
            "ItemIconName": "Open",
            "ItemIconOnly": false,
            "ItemKey": "commandOpen",
          },
        },
        "divider": undefined,
        "enabled": true,
        "farItem": undefined,
        "iconColor": "blue",
        "iconName": "Open",
        "iconOnly": false,
        "id": "1",
        "isHeader": undefined,
        "key": "commandOpen",
        "name": "Open",
        "overflow": undefined,
        "parentItemKey": undefined,
        "split": undefined,
        "topDivider": undefined,
        "visible": undefined,
      },
      Object {
        "checked": undefined,
        "data": MockEntityRecord {
          "getNamedReference": [MockFunction],
          "id": "2",
          "values": Object {
            "ItemDisplayName": "New",
            "ItemEnabled": false,
            "ItemIconColor": "blue",
            "ItemIconName": "New",
            "ItemIconOnly": true,
            "ItemKey": "commandNew",
          },
        },
        "divider": undefined,
        "enabled": false,
        "farItem": undefined,
        "iconColor": "blue",
        "iconName": "New",
        "iconOnly": true,
        "id": "2",
        "isHeader": undefined,
        "key": "commandNew",
        "name": "New",
        "overflow": undefined,
        "parentItemKey": undefined,
        "split": undefined,
        "topDivider": undefined,
        "visible": undefined,
      },
      Object {
        "checked": undefined,
        "data": MockEntityRecord {
          "getNamedReference": [MockFunction],
          "id": "3",
          "values": Object {
            "ItemDisplayName": "Save",
            "ItemEnabled": false,
            "ItemFarItem": true,
            "ItemIconColor": "green",
            "ItemIconName": "Save",
            "ItemIconOnly": true,
            "ItemKey": "commandSave",
          },
        },
        "divider": undefined,
        "enabled": false,
        "farItem": true,
        "iconColor": "green",
        "iconName": "Save",
        "iconOnly": true,
        "id": "3",
        "isHeader": undefined,
        "key": "commandSave",
        "name": "Save",
        "overflow": undefined,
        "parentItemKey": undefined,
        "split": undefined,
        "topDivider": undefined,
        "visible": undefined,
      },
      Object {
        "checked": undefined,
        "data": MockEntityRecord {
          "getNamedReference": [MockFunction],
          "id": "5",
          "values": Object {
            "ItemDisplayName": "Sub 1",
            "ItemIconName": "New",
            "ItemKey": "sub1",
            "ItemParentKey": "commandOpen",
          },
        },
        "divider": undefined,
        "enabled": undefined,
        "farItem": undefined,
        "iconColor": undefined,
        "iconName": "New",
        "iconOnly": undefined,
        "id": "5",
        "isHeader": undefined,
        "key": "sub1",
        "name": "Sub 1",
        "overflow": undefined,
        "parentItemKey": "commandOpen",
        "split": undefined,
        "topDivider": undefined,
        "visible": undefined,
      },
      Object {
        "checked": undefined,
        "data": MockEntityRecord {
          "getNamedReference": [MockFunction],
          "id": "6",
          "values": Object {
            "ItemDisplayName": "Sub 2",
            "ItemIconName": "Save",
            "ItemKey": "sub2",
            "ItemParentKey": "commandOpen",
          },
        },
        "divider": undefined,
        "enabled": undefined,
        "farItem": undefined,
        "iconColor": undefined,
        "iconName": "Save",
        "iconOnly": undefined,
        "id": "6",
        "isHeader": undefined,
        "key": "sub2",
        "name": "Sub 2",
        "overflow": undefined,
        "parentItemKey": "commandOpen",
        "split": undefined,
        "topDivider": undefined,
        "visible": undefined,
      },
    ]
  }
  onSelected={[Function]}
  setFocus=""
  tabIndex={5}
  themeJSON={null}
  visible={true}
  width={-1}
/>
`;
